flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NettyShuffleEnvironmentConfiguration.java

```diff
@@ -356,27 +354,9 @@ public static NettyShuffleEnvironmentConfiguration fromConfiguration(
         CompressionCodec compressionCodec =
                 configuration.get(NettyShuffleEnvironmentOptions.SHUFFLE_COMPRESSION_CODEC);
 
-        boolean batchShuffleCompressionEnabled;
-        if (compressionCodec == CompressionCodec.NONE) {
-            batchShuffleCompressionEnabled = false;
-        } else {
-            batchShuffleCompressionEnabled =
-                    configuration.get(
-                            NettyShuffleEnvironmentOptions.BATCH_SHUFFLE_COMPRESSION_ENABLED);
-
-            if (!batchShuffleCompressionEnabled) {
-                LOG.warn(
-                        "Deprecated configuration key {} is used to disable the compression. "
-                                + "Please set the {} to \"None\" to disable the compression.",
-                        NettyShuffleEnvironmentOptions.BATCH_SHUFFLE_COMPRESSION_ENABLED.key(),
-                        NettyShuffleEnvironmentOptions.SHUFFLE_COMPRESSION_CODEC.key());
-            }
-        }
+        boolean batchShuffleCompressionEnabled = compressionCodec != CompressionCodec.NONE;
 
-        int maxNumConnections =
-                Math.max(
-                        1,
-                        configuration.get(NettyShuffleEnvironmentOptions.MAX_NUM_TCP_CONNECTIONS));
+        int maxNumConnections = 1;
```

Comment:
Same with the last comment, if it's always 1, this parameter may be not needed, and some code can be simplified.